framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -328,6 +319,37 @@ public int runTests() throws Exception {
         }
       }
 
+      //To Determine whether to check for random failures or not depending on the number of failed tests being less than the number of threads
```

Comment:
Change comment to "/* Run all failed tests a second time. 
* Tests which succeed are categorized as random failures. 
* To prevent long runtimes, cmdParam.threads is chosen as an arbitrary max limit. 
* If total failed tests are higher than the max defined above, we do not isolate random failures. */"